RecCalorimeter/src/components/CreateCaloCells.cpp

```diff
@@ -67,7 +67,12 @@ StatusCode CreateCaloCells::initialize() {
   }
 
   // Copy over the CellIDEncoding string from the input collection to the output collection
-  m_cellsCellIDEncoding.put(m_hitsCellIDEncoding.get());
+  std::string hitsEncoding = m_hitsCellIDEncoding.get({});
+  if (hitsEncoding == "") {
+    error () << "Missing cellID encoding for input collection" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  m_cellsCellIDEncoding.put(hitsEncoding);
```

Comment:
The result of this change doesn't really change the outcome of all of this, IIUC.

```cpp
m_cellsCellIDEncoding.put(m_hitsCellIDEncoding.get());
```
will throw an exception in case there is no cellID Encoding string and Gaudi will stop. The current implementation effectively only exchanges the error message you get before a stop, right?

Anyhow, this will give you a slightly more interpretable error message, the only thing that could be improved here is to use the `get_optional` interface of the `MetaDataHandle`:

```suggestion
  auto hitsEncoding = m_hitsCellIDEncoding.get_optional();
  if (!hitsEncoding.has_value()) {
    error () << "Missing cellID encoding for input collection" << endmsg;
    return StatusCode::FAILURE;
  }
  m_cellsCellIDEncoding.put(hitsEncoding.value());
```
This makes it slightly more clear that you are doing the check on your end, rather then letting something else do it.

Reply:
I am doing get({}) so the code should not raise an exception. But if you think it is more readable if I use get_optional instead I will do this change after I get off my airplane (the flight is taking off in minutes). And I decided to return error/failure rather than a warning so that we can catch this and fix it in the future. At the moment we can run without problems because we can obtain the same results scheduling the tools in a different way that does not need to run this alg on collections without encoding

Reply:
If you just want the failure and are OK with the exception message that you get from the `MetaDataHandle` these changes can also be removed completely. The end result will be the same, the major difference will be the error message you get in the output.

For the `get_optional`, I would mainly like it because it would be the intended way to do this check. The current way with getting an empty default value and comparing against that also works in this case, but it might not in other cases, where the empty default value might be a valid value so the comparison will no longer work. In the end, here it is more about starting off with a good example, so that people who come across this, see how it "should" be done.

Reply:
implemented the suggestion to use get_optional in the latest commit